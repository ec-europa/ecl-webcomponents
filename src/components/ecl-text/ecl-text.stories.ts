const getArgs = () => {
  return {
    tag: 'div',
    level: 5,
    isBold: false,
    size: 'm',
    text: 'The quick brown fox jumps over the lazy dog',
  };
};

const getArgTypes = () => {
  return {
    tag: {
      type: { name: 'select' },
      options: ['div', 'p', 'h1' ,'h2', 'h3', 'h4', 'h5', 'h6'],
      description: 'Tag to render',
    },
    level: {
      type: { name: 'select' },
      options: [1, 2, 3, 4, 5, 6],
      description: 'Font size (not for paragraphs)',
    },
    isBold: {
      type: { name: 'boolean' },
      description: 'Higher font weight',
    },
    size: {
      type: { name: 'select' },
      options: ['xs', 's', 'm', 'l'],
      description: 'Shortcut to set font size (only for paragraphs)',
      table: {
        defaultValue: { summary: 'm' },
      }
    },
    text: {
      type: { name: 'string' },
      description: 'The visible text',
    },
  };
};

export default {
  title: 'Components/text',
};

const Template = args => {
  let story = `<ecl-text
    tag="${args.tag}"
    theme="${args.theme}"
    level="${args.level}"
    isBold=${args.isBold}
    size="${args.size}"
  >
    ${args.text}
  </ecl-text>`;

  return story;
}


export const Text = Template.bind({});
Text.storyName = 'default';
Text.args = getArgs();
Text.argTypes = getArgTypes();