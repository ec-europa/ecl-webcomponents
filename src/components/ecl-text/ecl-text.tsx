import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-text',
  styleUrls: {
    ec: './build/styles/ecl-text-ec.css',
    eu: './build/styles/ecl-text-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclText {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() tag: 'div' | 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span' | undefined;
  @Prop() theme: string = 'ec';
  @Prop() isBold: boolean = false;
  @Prop() level: 1 | 2 | 3 | 4 | 5 | undefined;
  @Prop() size: 'l' | 'm' | 's' | 'xs' | undefined;
  @Prop() itemId: string;

  getClass(): string {
    let styleClasses = [this.styleClass];

    if (this.level && !(this.size && this.tag === 'p')) {
      switch(this.level) {
        case 1:
          styleClasses.push('ecl-u-type-3xl');
          break;
    
        case 2:
          styleClasses.push('ecl-u-type-2xl');
          break;

        case 3:
          styleClasses.push('ecl-u-type-xl');
          break;

        case 4:
          styleClasses.push('ecl-u-type-prolonged-l');
          break;

        case 5:
          styleClasses.push('ecl-u-type-prolonged-m');
          break;

         default:
      }
    }

    if (this.size && this.tag === 'p') {
      styleClasses.push(`ecl-u-type-prolonged-${this.size}`);
    }

    if (styleClasses.length === 0) {
      styleClasses.push('ecl-u-type-prolonged-m');
    }

    if (this.isBold) {
      styleClasses.push('ecl-u-type-bold');
    }

    return styleClasses.join(' ');
  }

  render() {
    const Tag = this.tag;
    return (
      <Tag class={this.getClass()} id={this.itemId ? this.itemId : null}>
        <slot></slot>
      </Tag>
    )
  }
}

