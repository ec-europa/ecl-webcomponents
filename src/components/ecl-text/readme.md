# ecl-text



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type                                                                     | Default     |
| ------------ | ------------- | ----------- | ------------------------------------------------------------------------ | ----------- |
| `isBold`     | `is-bold`     |             | `boolean`                                                                | `false`     |
| `itemId`     | `item-id`     |             | `string`                                                                 | `undefined` |
| `level`      | `level`       |             | `1 \| 2 \| 3 \| 4 \| 5`                                                  | `undefined` |
| `size`       | `size`        |             | `"l" \| "m" \| "s" \| "xs"`                                              | `undefined` |
| `styleClass` | `style-class` |             | `string`                                                                 | `''`        |
| `tag`        | `tag`         |             | `"div" \| "h1" \| "h2" \| "h3" \| "h4" \| "h5" \| "h6" \| "p" \| "span"` | `undefined` |
| `theme`      | `theme`       |             | `string`                                                                 | `'ec'`      |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
