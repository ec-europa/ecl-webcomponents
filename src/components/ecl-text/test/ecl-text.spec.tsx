import { newSpecPage } from '@stencil/core/testing';
import { EclText } from '../ecl-text';

describe('ecl-text', () => {
  it('renders correctly as an h2 with h3 font size', async () => {
    const page = await newSpecPage({
      components: [EclText],
      html: `<ecl-text tag="h2" level=3>The quick brown fox jumps over the lazy dog</ecl-text>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it('renders correctly as a paragraph with with bold font style', async () => {
    const page = await newSpecPage({
      components: [EclText],
      html: `<ecl-text tag="p" is-bold>The quick brown fox jumps over the lazy dog</ecl-text>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});