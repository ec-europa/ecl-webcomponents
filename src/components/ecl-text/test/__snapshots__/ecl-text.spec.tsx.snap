// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-text renders correctly as a paragraph with with bold font style 1`] = `
<ecl-text is-bold="" tag="p">
  <!---->
  <p class="ecl-u-type-bold">
    The quick brown fox jumps over the lazy dog
  </p>
</ecl-text>
`;

exports[`ecl-text renders correctly as an h2 with h3 font size 1`] = `
<ecl-text level="3" tag="h2">
  <!---->
  <h2 class="ecl-u-type-xl">
    The quick brown fox jumps over the lazy dog
  </h2>
</ecl-text>
`;
